cmd/snap-confine/seccomp-support.c

```diff
@@ -109,6 +109,24 @@ int sc_apply_seccomp_bpf(const char *filter_profile)
 	sc_must_snprintf(profile_path, sizeof(profile_path), "%s/%s.bin",
 			 filter_profile_dir, filter_profile);
 
+	// Wait some time for the security profile to show up. When
+	// the system boots snapd will created security profiles, but
+	// a service snap (e.g. network-manager) starts in parallel with
+	// snapd so for such snaps, the profiles may not be generated
+	// yet
+	int max_wait = 120;
+	if (getenv("SNAP_CONFINE_MAX_PROFILE_WAIT") != 0)
+		if (atoi(getenv("SNAP_CONFINE_MAX_PROFILE_WAIT")) > 0)
+			max_wait =
+			    atoi(getenv("SNAP_CONFINE_MAX_PROFILE_WAIT"));
+	struct stat buf;
```

Comment:
You can just use `access(profile_path, F_OK) == 0` for this with less code to write.